### PR TITLE
Fix load factor limits

### PR DESCRIPTION
**Pull request by donstim**
Fixes the load factor limits for determining when structural failure may occur in flight due to high g maneuvers.

## Summary of Changes

Changes the load factor limits and the safety factor applied to correspond to real airplane values. I don't know where or how Asobo came up with the original values, but they fixed 4 out of 5 of them with their latest update. This PR adds the Asobo changes plus fixes the remaining one (Flaps Up negative g-limit) that Asobo did not fully correct.
## References

Load factor limits from A320 Airplane Flight Manual
Screenshot (469)
Factor of safety required by the transport airplane aviation safety regulations (this is the FAA requirement; European, Canadian, Brazilian, etc. all the same)
Screenshot (471)

## Additional context

The load factor limits refer to the ability of the airplane structure to withstand aerodynamic loads generated during maneuvering flight. They are specified in terms of g-limits - the amount of g-force that the structure is capable of withstanding. If the airplane maneuver exceeds what the structure is capable of handling the structure will fail.

As you can see from the screenshot of the A320neo g-limits, there are limits for both increasing g (positive g-limits) and decreasing g (negative g-limits). And these are positive and negative g-limits differ with flaps up from with flaps (or slats) extended. 

These g-limits are what are referred to as limit loads. The airplane structure must be able to withstand limit loads without any permanent deformation of the structure or interfere with safe operations. The airplane must be able to withstand what are called ultimate loads without failure. Ultimate loads are limit loads increased by a factor of safety.  The required factor of safety is 1.5. Loads up to ultimate loads are what the airplane in MSFS should be able to withstand without suffering stress damage that ends the flight.

To determine the ultimate loads in terms of the g-loads, simply add 1.5g additional capability to each limit g-load. So, the g-loads that the A320neo should be able to withstand in MSFS without suffering stress damage that ends the flight are:
Flaps up positive g-limit: 4.0
Flaps up negative g-limit: -2.5
Flaps down positive g-limit: 3.5
Flaps up negative g-limit: -1.5

Discord username (if different from GitHub):donbikes#4084

## Testing instructions

NOTE 1:  There should not be a need for extensive testing or precise evaluation of this mod. It merely implements changes that Asobo has made to their own flight_model.cfg file plus corrects one remaining error. It does not involve any "tweaking," tuning, or adjustment; it involves values that are either right or wrong.
NOTE 2: The A320NEO fly-by-wire envelope protection system includes load factor protection. I don't know how well it is implemented in the vanilla airplane (it seems to work okay), but for sure after our own custom FBW system is implemented, these load factor limitations on the structure will only come into play if the load factor protection function is disabled or inoperative.
NOTE 3: It should be sufficient just to show that the new g-limits work correctly. Since Asobo changed the values in their latest update, you should not have to compare this PR mod to the unmodified version.
NOTE 4: Because you will probably exceed the g-limits during testing, you can expect to have to re-start the flight a number of times. Also, because g-forces change rapidly, it is difficult to know at exactly what g-force the excessive stress damage occurred.

1. Make sure you have "Aircraft Stress Damage" enabled in the MSFS "Options" "Assistance" "Failure & Damage menu.
2. Run the sim in Developer mode (in the MSFS "Options" "General" Developers" menu)
3. In the DevMode Options menu at the top of the screen, make sure to uncheck "Disable crashes." 
4. Under "DevMode," select "New Project." Click "OK" in the "Create New Project" window that pops up.
5. In the Tools menu, select "Aircraft Editor." 
6. In the Debug menu of the aircraft editor, select "Sim Info." In the sim info window that pops up, scroll down until you can see "G-Force." This is what you will use to see what g-force your maneuver is exerting on the airplane.
7. Review the ultimate g-load limits listed in the "Additional Context" section above. 
8. Get the airplane to a decent altitude for testing (e.g, above 5,000 feet).
9. In the overhead panel, turn off ELAC1 and SEC1 on the left side of the panel under FLT CTL and ELAC2, SEC2, and SEC3 on the right side of the panel under FLT CTL. Without doing this the fbw system will limit the load factor you can reach.
10. Now for the testing!. For the positive g-limits, to be able to maximize the g-force, start at a high speed (up to VMO/VFE or even higher) in a shallow dive. With auto-pilot and autothrust off and with thrust levers in the TOGA detent, pull back on the joystick fairly abruptly to generate a high positive g-force. You will soon learn what it takes to generate the desired g-force. (It may be difficult to exceed the 4g positive ultimate g-load limit for the flaps up condition.) Because the maneuvers are so dynamic, it is difficult to verify the exact g-force at which MSFS stops the flight.  (Sometimes a momentary exceedance will lead to ending the flight due to structural damage in MSFS ; other times momentary exceedances don't end the flight.) You should aim for a maximum g-load of about 0.5g or so less than the ultimate limit to verify that the airplane can handle that much g, and then try to go over it and show that the stress damage limit is exceeded. It can be difficult to exceed the g-force limits. 
11. The highest negative g-forces are generally obtained by pushing forward on the stick with autopilot off and thrust levers at idle. 

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by donstim** [comment]
While this PR will correct the load factor limit values in the flight_model.cfg file, I have not been able to verify they make any difference. I have "aircraft stress causes damage" turned on the MSFS options, but I have been unable to get it to have any effect on any airplane. I've set the max load factor to 1.0 for the A320, pulled over 4g's (with the FBW system turned off) and not seen any effect. I've also pulled 4g's with the CJ4 and not seen any effect.

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 1. [FLIGHTMODEL] Preliminary fix for flaps 1 unrecoverable dives - @donstim (donbikes#4084)
 1. [UTILS] Added Vfe and Vfen table - @MisterChocker (Leon)
 1. [PFD] Fixed wrong Vfe and Vfen indication - @MisterChocker (Leon)
+1. [FLIGHTMODEL] Fix load factor limits - @donstim (donbikes#4084)
```
Comment (MisterChocker) [inline .github/CHANGELOG.md:19]:
```suggestion
1. [FLIGHTMODEL] Fixed load factor limits - @donstim (donbikes#4084)
```

**Review by MisterChocker — CHANGES_REQUESTED** [review]

File: .github/CHANGELOG.md (hunk 1 above)
Comment (donstim) [reply .github/CHANGELOG.md]: This change was made.

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg [hunk 2]
```diff
@@ -192,11 +192,11 @@ spoiler_extension_time = 2 ; Spoilers extension time (SECONDS)
 spoiler_handle_available = 1 ; Spoiler handles available true/false
 auto_spoiler_available = 1 ; Auto spoilers available true/false
 auto_spoiler_min_speed = 72 ; Auto spoilers minimum trigger speed
-positive_g_limit_flaps_up = 2.4 ; Flap positive load limit when up. Same dimension as gravity vector FEET/SECONDS^2
-positive_g_limit_flaps_down = 1.68 ; Flap positive load limit when down. Same dimension as gravity vector FEET/SECONDS^2
-negative_g_limit_flaps_up = -1.92 ; Flap negative load limit when up. Same dimension as gravity vector FEET/SECONDS^2
-negative_g_limit_flaps_down = -1.44 ; Flap negative load limit when down. Same dimension as gravity vector FEET/SECONDS^2
-load_safety_factor = 1 ; Flap negative load limit when down. Same dimension as gravity vector FEET/SECONDS^2
+positive_g_limit_flaps_up = 2.5 ; Flap positive load limit when up. Same dimension as gravity vector FEET/SECONDS^2
```
Comment (tracernz) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg:195]: If these are actually feet/seconds^2 as indicated they'd be quite a bit higher than you've got here (1g = 9.8 m/s2 = 32 ft/s2). Seems the unit in these comments is bogus?

Comment (donstim) [reply A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg:195]: The units are correct, but not really relevant to the issue.  Why do you think they are bogus?

**Review by MisterChocker — APPROVED** [review]
LGTM

**Review by Benjozork — APPROVED** [review]
LGTM

**Comment by ImenesFBW** [comment]
Name: Imenes#8739
Date of testing: Current date 10/01/2021
Version of the sim: 1.12.13.0
PR Tested: #2748
Object of test: Testing adjusted load factor limits 
Testing technique: Try-out (multiple tests)

## Steps you've made to test the Pull Request:

 * Started in the air
 * Enabled stress damage and popped up Sim Info to monitor the g-loads
 * Disabled ELAC1, SEC1, ELAC2, SEC2, and SEC3 
 * Performed high and low G maneuvers to try and over-G the airframe

### Flight notes

Testing this is a bit tricky as mentioned in the instructions. I tried keeping the g-load below 0.5 of max to confirm it does not overstress too early and then jank the stick to cause overstress. During the tests, it did sometimes seem that the sim would detect stress damage a tad early especially when testing flaps up positive g-limit, but only in rare cases. In general, all the limits were correct for a given situation.

## Overall Rating: Good

#### Conclusions of the testing:
 * LGTM! The values for stress damage seem to be working as described and documented.

**Comment by HugoTro** [comment]
Name: Hugo #3307
Date of testing: Current date 10/01/2021
Version of the sim: 1.12.13.0
PR Tested: #2748 

Testing technique: Over-G
Steps you've made to test the Pull Request:
1. Spawned on a runway
2. Climbed high and went 350 kts
3. Pushed on the stick to have -1.5G on the debug panel
4. Pulled on the stick 
5.

Overall Rating: Meh
Conclusions of the testing: See attached video, limits aren't being really respected by MSFS and I had to go in the 4G for it to show the overstress message [https://youtu.be/BWg5nOk6VOc](url)

**Comment by donstim** [comment]
> 
> Overall Rating: Meh
> Conclusions of the testing: See attached video, limits aren't being really respected by MSFS and I had to go in the 4G for it to show the overstress message [https://youtu.be/BWg5nOk6VOc](url)

Thanks for testing!! Love your rating! Kind of sums up my reaction to MSFS's stress damage feature, too.. I could not access your video, but 4g's is the right level for showing the overstress message (+2.5g load limit + 1.5G factor of safety).

**Comment by MisterChocker** [comment]
> I could not access your video

For some strange reason you need to copy the link and paste it into the address bar yourself, otherwise it redirects to github.

**Pull request merged**